frontend/src/metabase/visualizations/visualizations/Text/Text.unit.spec.tsx

```diff
@@ -153,6 +153,24 @@ describe("Text", () => {
         userEvent.click(screen.getByTestId("editing-dashboard-text-preview"));
         expect(screen.getByDisplayValue("text text text")).toBeInTheDocument();
       });
+
+      it("should call onUpdateVisualizationSettings on blur", () => {
+        const mockOnUpdateVisualizationSettings = jest.fn();
+        const options = {
+          settings: getSettingsWithText("text"),
+          isEditing: true,
+          onUpdateVisualizationSettings: mockOnUpdateVisualizationSettings,
+        };
+        setup(options);
+
+        userEvent.click(screen.getByTestId("editing-dashboard-text-preview"));
+        userEvent.type(screen.getByRole("textbox"), "foo");
+        userEvent.tab();
+
+        expect(mockOnUpdateVisualizationSettings).toHaveBeenCalledWith({
+          text: "textfoo",
+        });
+      });
```

Comment:
If `onUpdateVisualizationSettings` can have major performance impact, it could be beneficial to assert its only called once here